src/api/api.ts

```diff
@@ -1 +1,45 @@
-export {};
+import axios from 'axios';
+import Config from 'react-native-config';
+
+const YelpApi = axios.create({
+  baseURL: Config.API_URL,
+});
+
+YelpApi.interceptors.request.use(
+  function (config) {
+    config.params = {
+      accept: 'application/json',
+      ...config.params,
+    };
+    config.headers.Authorization = `Bearer ${Config.TOKEN}`;
+    return config;
+  },
+  function (error) {
+    return Promise.reject(error);
+  },
+);
+
+YelpApi.interceptors.response.use(
+  function (response) {
+    return response.data;
+  },
+  function (error) {
+    return Promise.reject(error);
+  },
+);
+
+export function GetApi<T>(
+  location: string,
+  sort: string,
+  limit: number,
+): Promise<T> {
+  return YelpApi.get(
+    `/search?location=${location}&sort_by=${sort}&limit=${limit}`,
+  )
+    .then(response => {
+      return response;
+    })
+    .catch(err => {
+      return err;
```

Comment:
Handle common errors and display alert in case screen wants you to display an alert